components/layout/block.rs

```diff
@@ -349,7 +347,7 @@ impl CandidateBSizeIterator {
                 block_container_block_size.scale_by(percent)
             }
             (LengthOrPercentage::Calc(calc), Some(block_container_block_size)) => {
-                calc.length() + block_container_block_size.scale_by(calc.percentage())
+                calc.resolve(block_container_block_size)
             }
             (LengthOrPercentage::Calc(calc), None) => calc.length(),
```

Comment:
You need to remove this because this is wrong, which is the reason we introduce the new method.